src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Formatting/DefaultRazorFormattingService.cs

```diff
@@ -125,13 +125,23 @@ internal class DefaultRazorFormattingService : RazorFormattingService
                 result = await pass.ExecuteAsync(context, result, cancellationToken);
             }
 
+            var originalText = context.SourceText;
             var edits = result.Edits;
             if (collapseEdits)
             {
-                var collapsedEdit = MergeEdits(result.Edits, context.SourceText);
+                var collapsedEdit = MergeEdits(result.Edits, originalText);
                 edits = new[] { collapsedEdit };
             }
 
+            // Make sure the edits actually change something, or its not worth responding
+            var textChanges = edits.Select(e => e.AsTextChange(originalText));
+            var changedText = originalText.WithChanges(textChanges);
+
+            if (changedText.ContentEquals(originalText))
```

Comment:
Is this expensive or optimized for only looking at the location where the collapsed edit touched?

Reply:
This is comparing the entire text, not just the where the edit touched, but it will try to use a checksum if it can. Iterating through all of the edits and getting the subtext for each, and comparing would definitely allocate more than this approach, even if this can't use the checksum, because it re-uses a small-ish pool char array as it goes through the texts.

I can benchmark it if you want, but compared to the actual formatting, which does the `WithChanges` call and diffs source etc. I don't think this would show up at all.

Reply:
> I don't think this would show up at all.

Sounds good enough to me! 😄